### ISSUE TITLE
Support IPv6

### ISSUE THREAD

**Issue by plett** [checkly/public-roadmap#132]
**Is your feature request related to a problem? Please describe.**
I run some dual stack IPv4 and IPv6 services, but Checkly is only able to connect to the services via IPv4, leading to incomplete monitoring.

**Describe the solution you'd like**
The ability to select `IPv4 only`, `IPv6 only` or `Dual Stack` in a check. It would be nice if `Dual Stack` alternated between IPv4 and IPv6 in the same way that scheduled checks rotate around multiple locations.

**Comment by tnolet** [comment]
@plett thanks for contributing this. Can you give a more detailed run down on what actually happens? We are using popular, well supported frameworks under the hood, so we are slightly dependant on that. However, can you let me know — just in on or two bullet points here — what the experience for you looks like in Checkly when we target your IPv6 stack?

**Comment by plett** [comment]
@tnolet Thanks for getting back to me on this.

I've come up with the following scenarios which use public Google sites, which should make it easy to reproduce later:

- When I give Checkly a URL with a hostname that has both A and AAAA records (eg `https://google.com/` ), it will connect to the IPv4 address(es) in the A record and successfully test it, but it will never attempt to connect to the IPv6 address.
- When I give Checkly a hostname which only has an IPv6 AAAA record (eg `https://ipv6.google.com/` ), the check fails with a request error: `Error: getaddrinfo ENOTFOUND ipv6.google.com`
- When I give Checkly a URL with a numeric IPv6 address in it (eg `https://[2a00:1450:4009:817::200e]/ `(which is what ipv6.google.com currently resolves to for me)), I get a different request error: `Error: connect EAFNOSUPPORT 2a00:1450:4009:817::200e:443 - Local (undefined:undefined)`.

Let me know if you need any more details on this.

**Comment by tnolet** [comment]
@plett just to be 100000% sure, this is about API checks, correct? Not Browser checks?

**Comment by plett** [comment]
> @plett just to be 100000% sure, this is about API checks, correct? Not Browser checks?

Yes, API checks. I just re-read my previous comments and noticed that I didn't explicitly say what type of check I was making - sorry!

That said, the same issue does exist with Browser checks. It wasn't something I had tested until just now, but the following Browser check script:
```
const { chromium } = require("playwright")
const browser = await chromium.launch()
const page = await browser.newPage()
await page.goto("https://ipv6.google.com/")
await browser.close()
```
errors with:
`ERROR - UserScript page.goto: net::ERR_NAME_NOT_RESOLVED at https://ipv6.google.com/`

**Comment by ndom91** [comment]
Hey guys,

So we run our checks in aws lambdas and I did a bit of digging and it turns out they do not support ipv6 at all. So we're dependant on amazon bringing support there first :+1:

**Comment by plett** [comment]
@ndom91 Thanks for the update - that would certainly explain it. For some reason I thought you were running in EC2s, but I've found the blog post now where you talk about how and why you use Lambda https://blog.checklyhq.com/how-i-made-aws-lambda-work-for-my-saas/

I guess this issue needs to be parked until Amazon supports IPv6 in Lambda.
